src/commands/init.js

```diff
@@ -30,9 +31,9 @@ const templates = [{
  *
  * @returns {Promise} - Promise for the command.
  */
-export default function init({ parsedArguments, parsedOptions }) {
+export default function init({ parsedArguments, parsedOptions, directory }) {
     const { list, force } = parsedOptions.options;
-    const { name, template, version } = parsedArguments.arguments;
+    const { template, version } = parsedArguments.arguments;
```

Comment:
I think that will work. It will still create the new directory under the directory provided to `--directory`, just not directly in it.